sstables/storage.cc

```diff
@@ -409,7 +413,29 @@ future<> filesystem_storage::wipe(const sstable& sst, sync_dir sync) noexcept {
     }();
 
     try {
-        co_await remove_by_toc_name(name, sync);
+        auto new_toc_name = co_await make_toc_temporary(name, sync);
+        auto dir_name = parent_path(new_toc_name);
```

Comment:
what if `new_toc_name == ""`?

Reply:
It would step on ENOENT from remove_file below and would anyway ignore this error. But you're right, this behavior was not exactly like that before the patch